backend/app/api/src/helpers/UitpasNumberValidator.ts

```diff
@@ -0,0 +1,114 @@
+import { SimpleError } from '@simonbackx/simple-errors';
+import { UitpasTokenRepository } from './UitpasTokenRepository';
+
+export class UitpasNumberValidatorStatic {
+    async checkUitpasNumber(uitpasNumber: string) {
+        if (!STAMHOOFD.UITPAS_API_CLIENT_ID) {
+            throw new SimpleError({
+                code: 'uitpas_api_not_configured',
+                message: 'UiTPAS API is not configured',
+                human: $t(
+                    'UiTPAS is niet volledig geconfigureerd. Neem contact op met de beheerder.',
+                ),
+            });
+        }
+        const access_token = await UitpasTokenRepository.getAccessTokenFor(); // for nothing, means for platform
+        const baseUrl = 'https://api-test.uitpas.be'; // TO DO: Use the URL from environment variables
+
+        const url = `${baseUrl}/passes/${uitpasNumber}`;
+        const myHeaders = new Headers();
+        myHeaders.append('Authorization', 'Bearer ' + access_token);
+        const requestOptions = {
+            method: 'GET',
+            headers: myHeaders,
+        };
+
+        const response = await fetch(url, requestOptions).catch((error) => {
+            // Handle network errors
+            throw new SimpleError({
+                code: 'uitpas_unreachable',
+                message: `Error retrieving pass by UiTPAS number: ${error.message}`,
+                human: $t(
+                    `De verbinding met UiTPAS is mislukt, dus konden we je UiTPAS-nummer niet valideren. Probeer het later opnieuw.`,
+                ),
+            });
+        });
+        if (!response.ok) {
+            // Handle non-200 responses by attempting to parse the response to get a message from UiTPAS API
+            const json = await response.json().catch(() => {}); // Ignore JSON parsing errors -> we will throw a generic error below
+            if (
+                json
+                && json.endUserMessage
+                && json.endUserMessage.nl
+                && typeof json.endUserMessage.nl === 'string'
+                && json.title
+                && typeof json.title === 'string'
+            ) {
+                // If the response contains an endUserMessage, throw an error with that message
+                throw new SimpleError({
+                    code: 'error_retrieving_pass_by_uitpas_number',
+                    message: `UiTPAS API returned an error: ${json.title}`,
+                    human: json.endUserMessage,
+                });
+            }
+            throw new SimpleError({
```

Comment:
Hierboven mag je de voledige json ook nog loggen naar console.error zodat het in onze logs komt, waardoor we bugs/problemen kunnen debuggen

Reply:
Eventueel ook voor die mét user messages